src/test/scala/lenthall/collections/EnhancedCollectionsSpec.scala

```diff
@@ -19,4 +19,9 @@ class EnhancedCollectionsSpec extends FlatSpec with Matchers {
 
     intSet should be(Set(3))
   }
+
+  it should "find the first Int in a Set" in {
+    val objectSet = Set("hello", 3, None, "world")
+    objectSet.firstByType[Int] should be(Some(3))
```

Comment:
perhaps the test could include a second `Int` to prove the firstness?

Reply:
Come to think of it, "first" in a Set is a slightly curious notion